app/Services/Model/Imp/TagService.php

```diff
@@ -58,4 +59,35 @@ public function addTag(array $payload): Tag
           return $tag;
        });
     }
+
+    /**
+     *@inheritDoc
+     */
+    public function updateTag(int $tag_id, array $payload): Tag
+    {
+        return $this->tx_service->transaction(function () use ($tag_id, $payload){
+            $tag = $this->tag_repository->getById($tag_id);
+            if(is_null($tag) || !$tag instanceof Tag){
+                throw new EntityNotFoundException("tag {$tag_id} not found");
+            }
+
+            $tag->setTag(trim($payload['tag']));
```

Comment:
u need check here for dupes before add and throw a validation error

Reply:
done